From 9994ac6f9f137441dfc3d49a5c9b6f914166dd25 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 22 Jan 2018 11:09:56 +0100
Subject: [PATCH 02/10] avcodec/h264_slice: use the new SAR early when setting
 the decoder

If we don't do that get_format might not be called for a while and the proper
SAR not used.

See the sample mentioned here: https://trac.videolan.org/vlc/ticket/19435
---
 libavcodec/h264_slice.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libavcodec/h264_slice.c b/libavcodec/h264_slice.c
index 7e53e38cca..a4bf58af1c 100644
--- a/libavcodec/h264_slice.c
+++ b/libavcodec/h264_slice.c
@@ -1090,8 +1090,10 @@ static int h264_init_ps(H264Context *h, const H264SliceContext *sl, int first_sl
         || (non_j_pixfmt(h->avctx->pix_fmt) != non_j_pixfmt(get_pixel_format(h, 0))))
         must_reinit = 1;
 
-    if (first_slice && av_cmp_q(sps->vui.sar, h->avctx->sample_aspect_ratio))
+    if (first_slice && av_cmp_q(sps->vui.sar, h->avctx->sample_aspect_ratio)) {
         must_reinit = 1;
+        ff_set_sar(h->avctx, sps->vui.sar);
+    }
 
     if (!h->setup_finished) {
         h->avctx->profile = ff_h264_get_profile(sps);
-- 
2.45.1.windows.1

